VendordepJsonSpecification.md

```diff
@@ -0,0 +1,28 @@
+WPILib Vendor Dependency JSON File Format Specification
+===
+
+- version: 1.0
+
+Types
+---
+| Typename | JSON description |
+---
+| JavaArtifact | `{ groupId: String, ArtifactId: String, version: String }` |
+| JniArtifact | `{ groupId: String, ArtifactId: String, version: String, skipInvalidPlatforms: Boolean, isJar: Boolean, validPlatforms: String[] }` |
+| CppArtifact | `{ groupId: String, ArtifactId: String, version: String, libName: String, headerClassifier: String, sharedLibrary: Boolean, skipInvalidPlatforms: Boolean, binaryPlatforms: String[] }` |
+
+Keys
+---
+| Name | Type | Description | Optional |
+---
+| name | String | The name of the vendordep. | No. |
+| version | String | The version of the vendordep. | No. |
+| uuid | String | The unique identifier for the vendordep. Used to avoid replicating vendordeps in projects. | No. |
+| mavenUrls | String[] | An array of urls pointing to the root of the maven repository storing the vendordep. | No. |
+| extraGroupIds | String[] | | Yes. |
```

Comment:
I don't know what this is or if it's a valid key I got my specifications from https://github.com/wpilibsuite/native-utils/blob/main/src/main/java/edu/wpi/first/nativeutils/vendordeps/WPIVendorDepsExtension.java#L310 and the example json file

Reply:
Im also not sure if
```java
        public VendorDependency[] requires;
        public VendorDependency[] conflictsWith;
```
should be added in here along with the corresponding types